src/styles/main.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  max-width: calc((#{$day-size} * 7 + (6 * #{$gap})) + #{$padding} * 2);
+  padding: $padding;
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: $day-size;
+    width: $day-size;
+    background-color: #eee;
+    border: 1px solid black;
+    cursor: pointer;
+    transition:
+      transform 0.5s,
+      background-color 0.5s;
+  }
+  &__day:hover {
+    background-color: #ffbfcb;
+    transform: translateY(-20px);
+  }
+
+  @for $num from 1 through 31 {
+    & :nth-child(#{$num})::before {
+      content: '#{$num}';
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+      text-align: center;
+    }
```

Comment:
The font properties here should be extracted into a placeholder selector or a mixin for reusability and to avoid duplication if these styles are applied elsewhere.